x-pack/plugins/endpoint/public/applications/endpoint/mocks/app_context_render.tsx

```diff
@@ -45,7 +47,12 @@ export const createAppRootMockRenderer = (): AppContextTestRender => {
   const history = createMemoryHistory<never>();
   const coreStart = coreMock.createStart({ basePath: '/mock' });
   const depsStart = depsStartMock();
-  const store = appStoreFactory({ coreStart, depsStart });
+  const middlewareSpy = createSpyMiddleware();
+  const store = appStoreFactory({
+    coreStart,
+    depsStart,
+    additionalMiddleware: [middlewareSpy.actionSpyMiddleware],
```

Comment:
@oatkiller could you review and comment on this change?
This goes along with a change in the `mocks/app_context_render` (see further below) and enables us to inject the `actionSpyMiddleware` into the application store for testing purposes. 

Let me know your thoughts

Reply:
I generally think this makes sense. One thought though:

When looking at the `appStoreFactory`, I see these comments:

```
/**
   * Any additional Redux Middlewares
   * (should only be used for testing - example: to inject the action spy middleware)
   */
```

```
        // Additional Middleware should go last
        ...additionalMiddleware
```

Based on those, could we replace: 
```
additionalMiddleware?: Array<ReturnType<typeof substateMiddlewareFactory>>;
``` 
with:
```
actionSpyMiddleware?: WhateverTheTypeShouldBe
```
Let me know your thoughts

Reply:
Thanks for the comments. Yeah, that makes sense. I assume that would be (from our `types.ts`) - `ReturnType<MiddlewareFactory>` so that it is correctly typed for the dispatch signature.